packages/@jsii/go-runtime/jsii-experimental/api.go

```diff
@@ -6,22 +6,286 @@ import (
 )
 
 // JSII Kernel API
-// Requests
+
+// Override is a public interface implementing a private method `isOverride`
+// implemented by the private custom type `override`. This is embedded by
+// MethodOverride and PropertyOverride to simulate the union type of Override =
+// MethodOverride | PropertyOverride.
+type Override interface {
+	isOverride()
+}
+
+type override struct{}
+
+func (o override) isOverride() {
+	return
+}
+
+type MethodOverride struct {
+	override
+
+	Method *string `json:"method"`
+	Cookie *string `json:"cookie"`
+}
+
+type PropertyOverride struct {
+	override
+
+	Property *string
+	Cookie   *string
+}
+
+func isMethodOverride(value Override) bool {
+	_, ok := value.(MethodOverride)
+	return ok
+}
+
+func isPropertyOverride(value Override) bool {
+	_, ok := value.(PropertyOverride)
+	return ok
+}
```

Comment:
Will `MethodOverride` and `PropertyOverride` always be values, or could they be used as pointers?  I ask because the type asserts are checking for the value version of of the specific type of override.

Depending on how the overrides can be used, it might be possible for unexpected silent errors to make it through because the override was used as the "wrong" value type.

two way to check for this:

1.) Check for both pointer and value version of the override type.
2.) Force the usage of Property and Method to be pointers by implementing the `isOverride` method with a pointer receiver.

Reply:
Using first approach as discussed offline!